test/integration-tests/Seaport.test.ts

```diff
@@ -12,135 +12,247 @@ import {
+      await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value })
+      const ownerAfter = await cryptoCovens.ownerOf(params.offer[0].identifierOrCriteria)
+      const wethBalanceAfter = await weth.balanceOf(bob.address)
+
+      expect(ownerBefore.toLowerCase()).to.eq(params.offerer)
+      expect(ownerAfter).to.eq(bob.address)
+      expect(wethBalanceBefore.sub(wethBalanceAfter)).to.eq(value)
     })
-    alice = await ethers.getSigner(ALICE_ADDRESS)
-    permit2 = (await deployPermit2()).connect(alice) as Permit2
-    router = (await deployUniversalRouter(permit2)).connect(alice) as UniversalRouter
-    planner = new RoutePlanner()
-    cryptoCovens = COVEN_721.connect(alice) as ERC721
-  })
 
-  it('completes a fulfillAdvancedOrder type', async () => {
-    const { advancedOrder, value } = getAdvancedOrderParams(seaportOrders[0])
-    const params = advancedOrder.parameters
-    const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
-      advancedOrder,
-      [],
-      OPENSEA_CONDUIT_KEY,
-      alice.address,
-    ])
-
-    planner.addCommand(CommandType.SEAPORT, [value.toString(), calldata])
-    const { commands, inputs } = planner
-
-    const ownerBefore = await cryptoCovens.ownerOf(params.offer[0].identifierOrCriteria)
-    await expect(router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value })).to.changeEtherBalance(
-      alice,
-      value.mul(-1)
-    )
-    const ownerAfter = await cryptoCovens.ownerOf(params.offer[0].identifierOrCriteria)
-
-    expect(ownerBefore.toLowerCase()).to.eq(params.offerer)
-    expect(ownerAfter).to.eq(alice.address)
+    it('completes an advanced order where ERC20 is already approved', async () => {
+      // seaportOrders[2] is an order containing ERC20 (WETH) as consideration
+      const { advancedOrder, value } = getAdvancedOrderParams(seaportOrders[2])
+      const params = advancedOrder.parameters
+      const considerationToken = params.consideration[0].token
+
+      // A previous txn which approves the conduit to spend the router's consideration token balance
+      planner.addCommand(CommandType.APPROVE_ERC20, [considerationToken, 0])
+      let { commands, inputs } = planner
+      await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value: 0 })
+
+      const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        [],
+        OPENSEA_CONDUIT_KEY,
+        bob.address,
+      ])
+
+      const permit = {
+        details: {
+          token: weth.address,
+          amount: value,
+          expiration: 0, // expiration of 0 is block.timestamp
+          nonce: 0, // this is his first trade
+        },
+        spender: router.address,
+        sigDeadline: DEADLINE,
+      }
+      const sig = await getPermitSignature(permit, bob, permit2)
+
+      planner.addCommand(CommandType.PERMIT2_PERMIT, [permit, sig])
```

Comment:
You havent reset the `planner` - so the approve command is actually still in there. Just need to add a quick `planner = new RoutePlanner()` to reset the planner before the next tx!

Reply:
In the gas test too please! The gas still went down - because the second approve no longer does a dirty STORE. But you should see it drop a lot further with the reset 👀